Storage/ThirdParty/SwiftData.swift

```diff
@@ -350,6 +350,14 @@ private class SQLiteDBStatement {
             sqlite3_finalize(self.pointer)
         }
     }
+    
+    func makeUtf8CString(from str: String) -> UnsafeMutablePointer<Int8> {
+        let count = str.utf8CString.count
+        let result: UnsafeMutableBufferPointer<Int8> = UnsafeMutableBufferPointer<Int8>.allocate(capacity: count)
+        // func initialize<S>(from: S) -> (S.Iterator, UnsafeMutableBufferPointer<Element>.Index)
+        _ = result.initialize(from: str.utf8CString)
+        return result.baseAddress!
```

Comment:
What happens if the force unwrap fails?

Reply:
force unwrap failure crashes app. however, according to this discussion https://forums.swift.org/t/when-is-unsafemutablebufferpointer-baseaddress-nil/32136 , baseAddress is not nil if count of buffer is greater than 0. Given a valid String as input, the `str.utf8CString.count` is always greater than 0.

Reply:
After further testing with character conversion mentioned in [this crash](https://github.com/mozilla-mobile/firefox-ios/issues/8744), I found the root cause is that a string with invalid unicode codepoint causes `cString(using:)` function return nil. While the same string has non-nil return when using `utf8CString`.

When invalid unicode code point is sent to (Swift) String initializor, the invalid character is converted to U+FFFD (replacement character).

A testing playground gist is added [here](https://gist.github.com/kcome/cacc927d7453bbead38137ca0960ec75)